### PR TITLE
Feat: add docker compose

### PR DESCRIPTION
**Pull request by HugoPoi**
* Add a docker-compose.yml file tested !
* Add callback url for twitter in pre-setup README ( This cause me some probleme )
* Add .env to .gitignore

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -39,3 +39,15 @@ BLOCKED_DOMAINS=evil.corp,bad.instance
...
+# Database settings
+# Postgres
+DB_HOST=localhost
+DB_PORT=5432
+DB_NAME=mastodon_twitter_poster_production
+DB_USER=mastodon_twitter_poster
+DB_PASS=
+
+# Redis only for Sidekiq
+REDIS_PROVIDER=REDIS_URL
```
Comment (renatolond) [inline .env.example:52]: What is this used for?

**Comment by renatolond** [comment]
Hey, thanks for this one, it's great to have a docker compose :)
Hm, weird, I thought the image had psql, since it has postgresql-dev, but I'm seeing it comes with postgresql-client. I would prefer adding that in the image to simplify the setup step, since the db init script is rails standard and we can't change back to the ruby schema because of implementation specific fields (the enums used in most tables are not implemented by rails and so cannot be exported)

File: .env.example (hunk 1 above)
Comment (HugoPoi) [reply .env.example:52]: `DB_PASS` is for setting the password of postgresql connection.
`REDIS_PROVIDER=REDIS_URL` is for indicate that REDIS_URL is the env var for setting redis connection for sidekiq, I' m not sure there is a default for this in sidekiq.

**Comment by HugoPoi** [comment]
I will check about this psql init stuff this week, maybe tonight. I pretty sure `postgresql-dev` doesn't provide psql but only psql client library needed for ruby psql client library.

**Comment by HugoPoi** [comment]
Much better now !
* I add  `postgresql-client` in Dockerfile
* And I found a much better elegant way of init the db with just `docker-compose run --rm web bundle exec rake db:setup`

File: .env.example (hunk 1 above)
Comment (renatolond) [reply .env.example:52]: Yeah, the default is REDIS_URL, I think you can remove this one :)

Comment (HugoPoi) [reply .env.example]: Done !

**Comment by renatolond** [comment]
Looks good, thanks again!

**Pull request merged**